processing/src/main/java/org/apache/druid/segment/ColumnCapabilitiesProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment;
+
+import org.apache.druid.segment.column.ColumnCapabilities;
+
+import javax.annotation.Nullable;
+
+public interface ColumnCapabilitiesProvider
```

Comment:
How about `ColumnDescriptions` - segment columns are written to and loaded from a segment via a `ColumnDescriptor` which populates the `ColumnCapabilities` for segments, so it sort of makes sense I guess, and is generic enough to apply to things that aren't actual segments too. idk, naming things is the hardest part. `AdapterCapabilities`? Something with the word schema in it?

Reply:
`ColumnInspector` since it's sort of like `VectorSizeInspector`, a shared interface for a common pattern from things that aren't really exactly the same thing?

Reply:
ColumnInspector sounds good to me.